src/model/losses.py

```diff
@@ -0,0 +1,125 @@
+import torch
+import torch.nn as nn
+
+__all__ = [
+    'CrossEntropyLossWrapper',
+    'DiceLoss',
+    'TverskyLoss',
+]
+
+
+class CrossEntropyLossWrapper(nn.Module):
+    """The cross-entropy loss wrapper.
+    Args:
+        weight (sequence, optional): The weight given to each class (default: None).
+    """
+
+    def __init__(self, weight=None, **kwargs):
+        super().__init__()
+        if weight is not None:
+            weight = torch.tensor(weight, dtype=torch.float)
+            kwargs.update({'weight': weight})
+        self.loss_fn = nn.CrossEntropyLoss(**kwargs)
+
+    def forward(self, output, target):
+        """
+        Args:
+            output (torch.Tensor) (N, C, *): The output logits.
+            target (torch.LongTensor) (N, *): The target where each value is between 0 and C-1.
+
+        Returns:
+            loss (torch.Tensor) (0): The cross entropy loss.
+        """
+        return self.loss_fn(output, target)
+
+
+class DiceLoss(nn.Module):
+    """The Dice loss.
+    Refs:
+        https://github.com/pytorch/pytorch/issues/1249#issuecomment-337999895
+
+    Args:
+        smooth (float, optional): The smooth term (default: 1.0).
+        square (bool, optional): Whether to use the square of cardinality (default: True).
+        weight (sequence, optional): The weight given to each class (default: None).
+    """
+
+    def __init__(self, smooth=1.0, square=True, weight=None):
+        super().__init__()
+        self.smooth = smooth
+        self.square = square
+        if weight is not None:
+            weight = torch.tensor(weight, dtype=torch.float)
+            self.register_buffer('weight', weight)
+
+    def forward(self, output, target):
+        """
+        Args:
+            output (torch.Tensor) (N, C, *): The output probability.
```

Comment:
Not very sure. Maybe we can mimic the [CrossEntropyLoss](https://pytorch.org/docs/stable/nn.html#crossentropyloss) in Pytorch that accepts the output logits instead of the output probability.

Reply:
The same as the `BinaryCrossEntropy` (ref: https://pytorch.org/docs/stable/nn.functional.html?highlight=binary%20cross%20entropy#torch.nn.functional.binary_cross_entropy), we can implement the binary_dice_loss function.